### PR TITLE
Remove stream config

### PR DESCRIPTION
**Pull request by ocefpaf**
Start to address #99

### CODE REVIEW

**Comment by ocefpaf** [comment]
@kthyng sorry to ping you here but who would be the best person to review this one?

**Comment by kthyng** [comment]
@ocefpaf No problem! We're discussing internally about getting some people involved.

**Comment by iwensu0313** [comment]
Hi @ocefpaf - thanks for taking a stab here. @melanie and I took a look at it together. These changes to documentation look good. 

We did notice that `StreamConfig` is still referenced in the following places:
* [`ContextConfig` section of the Usage page](https://github.com/ioos/ioos_qc/blob/daba0eeb4aec729bbe7d0efb6c6370e2531e45be/docs/source/usage.rst#contextconfig)
* Comments and docstrings in `config.py` e.g. [`ContextConfig` class definition docstring](https://github.com/ioos/ioos_qc/blob/daba0eeb4aec729bbe7d0efb6c6370e2531e45be/ioos_qc/config.py#L369)

But `StreamConfig` objects don't seem to exist anymore in the most recent version of the `ioos_qc` package. Digging around to figure out what the replacement is. Perhaps just `Config`? Any ideas?

File: docs/source/usage.rst [hunk 1]
```diff
@@ -83,34 +82,6 @@ In addition, the ``ContextConfig`` and ``Config`` objects can be initialized wit
...
-                'fail_span': [0, 12],
-            }
-        }
-    }
-    c = StreamConfig(config)
-
-
 ContextConfig
 ~~~~~~~~~~~~~
 A ``ContextConfig`` object defines multiple ``StreamConfig`` objects as well as optional `region` and `window` objects.
```
Comment (iwensu0313) [inline docs/source/usage.rst:87]: Should `StreamConfig` be replaced w/ something else here? What's the new object called?

**Comment by ocefpaf** [comment]
> But `StreamConfig` objects don't seem to exist anymore in the most recent version of the `ioos_qc` package. Digging around to figure out what the replacement is. Perhaps just `Config`? Any ideas?

Quoting Kyle from https://github.com/ioos/ioos_qc/issues/99#issuecomment-1452241539:

> The docs need to be updated, the only interface you need to load any config object is the `Config` class: `from ioos_qc.config import Config`

My guess is that this is already documented, right? So it is a matter of just removing StreamConfig. I'll remove it from the places you pointed out.

File: docs/source/usage.rst (hunk 1 above)
Comment (ocefpaf) [reply docs/source/usage.rst]: ContextConfig can take any ConfigType if I understood the code correctly. I changed to that instead of StreamConfig.

Comment (iwensu0313) [reply docs/source/usage.rst]: Yea I see that now, `ConfigType` object is what we're passing when initializing `ContextConfig` class.

**Comment by iwensu0313** [comment]
Looks good to me! @ocefpaf

**Review by iwensu0313 — APPROVED** [review]

**Pull request merged**